typescript/sdk/src/ism/EvmIsmReader.ts

```diff
@@ -201,6 +202,24 @@ export class EvmIsmReader extends HyperlaneReader implements IsmReader {
 
     this.assertModuleType(await ism.moduleType(), ModuleType.ROUTING);
 
+    // check if its the ICA ISM
+    try {
+      const icaInstance = InterchainAccountRouter__factory.connect(
+        address,
+        this.provider,
+      );
+      await icaInstance.CCIP_READ_ISM();
+      return {
+        address,
+        type: IsmType.ICA,
+      };
```

Comment:
this would be where we could use the message context to continue deriving

Reply:
14040e159